gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/adapter/PortalNavigationItemAdapter.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.adapter;
+
+import io.gravitee.apim.core.portal_page.model.*;
+import java.util.HashMap;
+import java.util.Map;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface PortalNavigationItemAdapter {
+    PortalNavigationItemAdapter INSTANCE = Mappers.getMapper(PortalNavigationItemAdapter.class);
+
+    com.fasterxml.jackson.databind.ObjectMapper OBJECT_MAPPER = new com.fasterxml.jackson.databind.ObjectMapper();
+
+    default PortalNavigationItem toEntity(io.gravitee.repository.management.model.PortalNavigationItem portalNavigationItem) {
+        return switch (portalNavigationItem.getType()) {
+            case FOLDER -> portalNavigationFolderFromRepository(portalNavigationItem);
+            case PAGE -> portalNavigationPageFromRepository(portalNavigationItem);
+            case LINK -> portalNavigationLinkFromRepository(portalNavigationItem);
+        };
+    }
+
+    default PortalArea mapArea(io.gravitee.repository.management.model.PortalNavigationItem.Area area) {
+        return switch (area) {
+            case HOMEPAGE -> PortalArea.HOMEPAGE;
+            case TOP_NAVBAR -> PortalArea.TOP_NAVBAR;
+        };
+    }
+
+    default io.gravitee.repository.management.model.PortalNavigationItem.Area mapAreaReverse(PortalArea area) {
+        return switch (area) {
+            case HOMEPAGE -> io.gravitee.repository.management.model.PortalNavigationItem.Area.HOMEPAGE;
+            case TOP_NAVBAR -> io.gravitee.repository.management.model.PortalNavigationItem.Area.TOP_NAVBAR;
+        };
+    }
+
+    @Mapping(target = "id", expression = "java(PortalNavigationItemId.of(portalNavigationItem.getId()))")
+    @Mapping(target = "organizationId", source = "organizationId")
+    @Mapping(target = "environmentId", source = "environmentId")
+    @Mapping(target = "title", source = "title")
```

Comment:
is it required ? I thought mapstruct was able to map automatically field with same name